src/game_over.rs

```diff
@@ -29,13 +32,9 @@ fn lose(
     mut state: ResMut<State<GameState>>,
     mut score_events: EventWriter<ScoreEvent>,
 ) {
-    for event in events.iter() {
-        match event {
-            PlayerDeadEvent => {
-                state.set(GameState::GameOver).unwrap();
-                score_events.send(ScoreEvent("You lose".to_string()));
-            }
-        }
+    for PlayerDeadEvent in events.iter() {
```

Comment:
"infallible patterns", let you match on something if it can only be one thing. It's typically used with tuples, but also works with other type constructors.
```rust
//  vvvvvvvvv infallible pattern
for (i, elem) in array.iter().enumerate() {
//  vvvvvvvvvvvvvvv also infallible pattern
for MyStruct(value) in array_of_my_struct.iter() {